src/Core/Environment.php

```diff
@@ -187,16 +187,22 @@ public static function getTimeLimitMax()
      */
     public static function getEnv($name)
     {
-        switch (true) {
-            case  is_array(static::$env) && array_key_exists($name, static::$env):
-                return static::$env[$name];
-            case  is_array($_ENV) && array_key_exists($name, $_ENV):
-                return $_ENV[$name];
-            case  is_array($_SERVER) && array_key_exists($name, $_SERVER):
-                return $_SERVER[$name];
-            default:
-                return getenv($name);
+        if (array_key_exists($name, static::$env)) {
+            return static::$env[$name];
         }
+        // isset() is used here instead of array_key_exists() to fix a very strange issue that occured in
+        // CI running silverstripe/recipe-kitchen-sink where PHP would timeout due apparently due to an
+        // excessively high number of array method calls. isset() is not used for static::$env above because
+        // values there may be null, and isset() will return false for null values
+        // Symfony also uses isset() for reading $_ENV and $_SERVER values
+        // https://github.com/symfony/dependency-injection/blob/6.2/EnvVarProcessor.php#L148
```

Comment:
I think we don't need to explain that much detail here.
Just say `isset` is used to prevent performance issues with too many `array_*` method calls and that Symfony also uses this approach.

Reply:
I'm inclined to agree with this.

Reply:
I like my essay

I think it's important for future generations to explain exactly why there's the array_key_exists() for static::$env and the we're immediately inconsistent and less precise using isset() for $_ENV and $_SERVER.

It's caused such an utterly obscure issue I think it needs a big explanation.